### PR TITLE
First Attempt, Might be Buggy

### PR DESCRIPTION
**Pull request by kangraye**
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment? 
12 hours

Please fill in one error and/or error message you received while working on this assignment.
My counter variables seem to overflow from one function to another.

What part of the assignment did you spend the most time on?
Debugging.

Comfort Level (1-5):
3

Completeness Level (1-5):
4

What did you think of this deliverable?
Good.

Is there anything in this code that you feel pleased about?
Unsure about it.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -1,4 +1,569 @@
...
+  } else if (gameState == 1) {
+    return setName(input);
+  } else if (gameState == 2) {
+    return setBet(input);
+  } else if (gameState == 3) {
+    return playerState(input);
+  } else if (gameState == 4) {
+    return dealerState(input);
+  }
+};
```
Comment (emily-zhangg) [inline script.js:552]: good job using game states to switch into different modes but the naming of the game states could be clearer. It is not very clear to me what 1-4 represents

File: script.js [hunk 2]
```diff
@@ -1,4 +1,569 @@
...
+      return hitPlayer();
+    }
+  } else if (input == "Split") {
+    if (splitEligibility() == true) {
+      return splitPlayer();
+    } else {
+      outputValue += "You are unable to split with your current hand.";
+      return outputValue;
+    }
+  }
```
Comment (emily-zhangg) [inline script.js:445]: if a function has many ifs, how many ifs must there be if we were to say that there are too many ifs? Maybe we should 'switch' the approach

Comment (emily-zhangg) [reply script.js:445]: here's a link to the switch statement that you can use https://www.w3schools.com/js/js_switch.asp

File: script.js (hunk 1 above)
Comment (emily-zhangg) [reply script.js:552]: same comment as above regarding the use of too many ifs and else ifs

File: script.js [hunk 3]
```diff
@@ -1,4 +1,569 @@
...
+    var playerCredits = 100;
+    var currentHand = [];
+    var player = {
+      name: playerName,
+      credits: playerCredits,
+      hand: currentHand,
+      bet: 0,
+      score: [0],
+      blackjack: false,
+    };
```
Comment (emily-zhangg) [inline script.js:77]: good job organizing player data in objects

**Review by emily-zhangg — COMMENTED** [review]
great attempt overall! The commenting and sectioning of code into helper functions are really helpful to the reader. One possible future improvement is to reduce the nesting of if and else ifs in your code. If you find yourself needing multiple if statements, consider using a switch instead

File: script.js (hunk 2 above)
Comment (kangraye) [reply script.js:453]: Thanks! Didn't know this elegant method exists!

**Pull request closed without merging**